### ISSUE TITLE
Does `gdal_utils` support configuration options?

### ISSUE THREAD

**Issue by kadyb** [r-spatial/sf#2003]
Can I use [configuration options](https://gdal.org/user/configoptions.html) directly in `sf::gdal_utils()`? I mean something like this:

```r
library("sf")
in_file = system.file("shape/storms_xyz.shp", package = "sf")
out_file = tempfile(fileext = ".gpkg")
gdal_utils(
  util = "vectortranslate",
  source = in_file,
  destination = out_file,
  options = c("-f", "GPKG",
              "--config OGR_CURRENT_DATE 29-MAY-2000 13:14:03")
)
```

> Error in CPL_gdalvectortranslate(source, destination, options, oo, doo,  : 
>   vectortranslate: options error
> In addition: Warning message:
> In CPL_gdalvectortranslate(source, destination, options, oo, doo,  :
>   GDAL Error 6: Unknown option name '--config OGR_CURRENT_DATE 29-MAY-2000 13:14:03'

**Comment by agila5** [comment]
Hi! I'm not really familiar with these topics (in fact I just learned most of this stuff :sweat_smile: so excuse for any silly mistake), but I think you can use `Sys.setenv(...)`. For example: 

``` r
library("sf")
#> Linking to GEOS 3.11.1, GDAL 3.6.2, PROJ 9.1.1; sf_use_s2() is TRUE
library("RSQLite")
in_file = system.file("shape/storms_xyz.shp", package = "sf")
out_file = tempfile(fileext = ".gpkg")
Sys.setenv(OGR_CURRENT_DATE = "2020-01-02T03:04:05.678Z")
gdal_utils(
  util = "vectortranslate",
  source = in_file,
  destination = out_file,
  options = c("-f", "GPKG")
)
con <- dbConnect(SQLite(), out_file)
dbGetQuery(con, 'SELECT last_change FROM gpkg_contents')$last_change
#> [1] "2020-01-02T03:04:05.678Z"
```

<sup>Created on 2023-02-20 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>


See also https://github.com/r-spatial/sf/issues/1618 for more details (i.e. how to specify the date, its format...) on several parts of this example.

**Comment by edzer** [comment]
Thanks! As `st_write` received a `config_options` argument (#1618), we may want to do this in a similar way for `gdal_utils`.

**Comment by edzer** [comment]
Like this:
```r
library("sf")
# Linking to GEOS 3.11.1, GDAL 3.6.2, PROJ 9.1.1; sf_use_s2() is TRUE
library("RSQLite")
in_file = system.file("shape/storms_xyz.shp", package = "sf")
out_file = tempfile(fileext = ".gpkg")
gdal_utils(
  util = "vectortranslate",
  source = in_file,
  destination = out_file,
  options = c("-f", "GPKG"),
  config_options = c(OGR_CURRENT_DATE = "2020-01-02T03:04:05.678Z")
)
con <- dbConnect(SQLite(), out_file)
dbGetQuery(con, 'SELECT last_change FROM gpkg_contents')$last_change
# [1] "2020-01-02T03:04:05.678Z"
```

**Comment by kadyb** [comment]
Thank you!

CC: @paul-carteron

**Comment by agila5** [comment]
Thanks for the quick fix and implementation!

**Comment by edzer** [comment]
Cc: @JoshOBrien

**Comment by JoshOBrien** [comment]
@edzer Thanks! That's terrific.

I've added support for `config_options` in a local branch of *gdalUtilities*, and will push to GitHub and then submit a new version to CRAN once this enhancement in *sf* makes it to CRAN.

**Comment by edzer** [comment]
now on CRAN.

**Comment by JoshOBrien** [comment]
@edzer Thanks for the notification. I think I'll wait until the next release of *sf*, when the fix in 73b2ea3 makes it to CRAN, as for now the examples in my package that rely on `"gdaldemprocessing"` fail.
